src/document_basic.c

```diff
@@ -175,6 +243,18 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx, QueryError
   size_t nitems = sctx->spec->numFields;
   JSONResultsIterator jsonIter = NULL;
 
+  RedisModuleKey *k = RedisModule_OpenKey(sctx->redisCtx, doc->docKey, REDISMODULE_READ);
```

Comment:
Why open and close the key again, instead of calling `getDocExpirationTime` when the key is already open below, i.e., under line 258 in which the key is opened via the json API?

Reply:
What we get at line 258 is `RedisJSON` object, not a `RedisModuleKey` object.